imageio/plugins/tifffile.py

```diff
@@ -175,6 +175,12 @@
 try:
     import tifffile as _tifffile
 except ImportError:
+    warnings.warn(
+        "ImageIO's vendored tifffile backend is deprecated and will be"
+        " removed in ImageIO v3. Install the tifffile directly:"
+        " `pip install imageio[tifffile]`", 
+        DeprecationWarning
+    )
```

Comment:
Dito, under what circumstances does this warning occur?

Reply:
This will occur if a user loads a TIFF file and doesn't have `tifffile` installed. 

I'm not 100% sure if we are making `tifffile` a primary dependency once we remove this. You will probably say we should, and the bioinformatics community will use it a lot, but for other scenarios, it is rather optional. In most cases it won't hurt to have it anyway but considering that there are use-cases with space concerns (e.g. the recently discussed freezing of apps) it might be better to keep things that can be optional optional.